### PR TITLE
feat: quiz button, popups, disabled mining

### PR DESCRIPTION
**Pull request by ice-orion**

### CODE REVIEW

File: src/api/kyc/types.ts [hunk 1]
```diff
@@ -20,3 +20,13 @@ export type Question = {
...
 };
+
+export type QuizStatus = {
+  kycQuizAvailabilityStartedAt: string;
+  kycQuizAvailabilityEndedAt: string;
+  kycQuizAvailable: boolean;
+  kycQuizCompleted: boolean;
+  kycQuizDisabled: boolean;
+  kycQuizRemainingAttempts?: number;
+  kycQuizResetAt: string[];
```
Comment (ice-ares) [inline src/api/kyc/types.ts]: u don't need this

File: src/api/tokenomics/types.ts [hunk 2]
```diff
@@ -34,6 +34,7 @@ export type MiningSummary = {
   remainingFreeMiningSessions?: number;
   availableExtraBonus?: number;
   miningStarted?: boolean;
+  kycStepBlocked?: number;
```
Comment (ice-ares) [inline src/api/tokenomics/types.ts]: remove this

File: src/screens/HomeFlow/Home/hooks/useHomeRefresh.ts [hunk 3]
```diff
@@ -31,6 +32,7 @@ const REFRESH_ACTIONS = [
     },
   },
   AchievementsActions.LEVELS_AND_ROLES_LOAD,
+  QuizActions.CHECK_QUIZ_STATUS,
```
Comment (ice-ares) [inline src/screens/HomeFlow/Home/hooks/useHomeRefresh.ts:35]: is this being called every 10 secs?

Comment (ice-orion) [reply src/screens/HomeFlow/Home/hooks/useHomeRefresh.ts:35]: nope, this one is for pull to refresh

File: src/api/tokenomics/types.ts (hunk 2 above)
Comment (ice-orion) [reply src/api/tokenomics/types.ts]: done

File: src/api/kyc/types.ts (hunk 1 above)
Comment (ice-orion) [reply src/api/kyc/types.ts]: done

File: src/store/modules/Quiz/sagas/resetQuizKycStep.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import {hasUnfinishedQuizSelector} from '@store/modules/Quiz/selectors';
+import {getErrorMessage} from '@utils/errors';
+import {call, put, SagaReturnType, select} from 'redux-saga/effects';
+
+type Action = ReturnType<
+  | typeof AppCommonActions.APP_INITIALIZED.SUCCESS.create
+  | typeof QuizActions.RESET_QUIZ_KYC_STEP.START.create
+>;
+
+export function* resetQuizKycStepSaga(action: Action) {
```
Comment (ice-ares) [inline src/store/modules/Quiz/sagas/resetQuizKycStep.ts:20]: when is this called?

File: src/store/modules/Tokenomics/selectors/index.ts [hunk 5]
```diff
@@ -82,3 +89,7 @@ export const getTotalCoinsStatsSelector =
 export const totalCoinsSelector = (state: RootState) => {
   return state.tokenomics.totalCoins.total;
 };
+
+export const kycStepBlockedSelector = (state: RootState) => {
+  return state.tokenomics.miningSummary?.kycStepBlocked;
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/selectors/index.ts]: remove this

File: src/store/modules/Quiz/sagas/resetQuizKycStep.ts (hunk 4 above)
Comment (ice-orion) [reply src/store/modules/Quiz/sagas/resetQuizKycStep.ts:20]: 1. When user interrupts a quiz without completion - by pressing the back button and confirming the warning
2. When user starts a quiz (set hasUnfinishedQuiz to true), closes the app without quiz completion, opens the app again

File: src/store/modules/Tokenomics/selectors/index.ts (hunk 5 above)
Comment (ice-orion) [reply src/store/modules/Tokenomics/selectors/index.ts]: done

**Review by ice-ares — APPROVED** [review]

**Review by ice-hades — APPROVED** [review]

**Review by ice-hector — APPROVED** [review]

**Pull request merged**